### ISSUE TITLE
BatchNorm中，若batchSIze为1的输入，会怎么处理

### ISSUE THREAD

**Issue by zehua9500** [PaddlePaddle/Paddle#19551]
训练时，BatchNorm中，若batchSize为1的输入，会怎么处理。按照公式的话均值是他自身，减完自身不就变成全0了吗。若是保持不变的话，那inference时减均值后就和训练时的数据完全不同了

**Comment by luotao1** [comment]
batch_norm的[API文档](https://old.paddlepaddle.org.cn/documentation/docs/en/1.5/api/layers/nn.html#batch-norm)
![image](https://user-images.githubusercontent.com/6836917/63991733-7d677b00-cb1b-11e9-89d4-bd9c4761bb19.png)

mean和var是momentum更新的，momentum默认是0.9，所以只要过了一轮后，mean就不会是X了。

batch norm原理是先白化再scale + offset，所以batchsize=1的时候就是应该白化到0，然后scale+offset，scale+offset是可学习的，而训练过程中会滚动更新mean和var，infer的时候用的是mean和var，所以没有影响

**Comment by zehua9500** [comment]
> batch_norm的[API文档](https://old.paddlepaddle.org.cn/documentation/docs/en/1.5/api/layers/nn.html#batch-norm)
> ![image](https://user-images.githubusercontent.com/6836917/63991733-7d677b00-cb1b-11e9-89d4-bd9c4761bb19.png)
> 
> mean和var是momentum更新的，momentum默认是0.9，所以只要过了一轮后，mean就不会是X了。
> 
> batch norm原理是先白化再scale + offset，所以batchsize=1的时候就是应该白化到0，然后scale+offset，scale+offset是可学习的，而训练过程中会滚动更新mean和var，infer的时候用的是mean和var，所以没有影响

恩好的谢谢，因为文档上的公式上写的均值为mini-batch的均值，所以有点迷惑，实际用的过程中感觉应该是用momentum而不是只看mini-batch，谢谢！

**Comment by luotao1** [comment]
好的，我们会增强下文档上的公式，感谢反馈！
